src/Bicep.Core.Samples/Files/AKS_LF/main.syntax.bicep

```diff
@@ -1,392 +1,393 @@
 // mandatory params
-//@[19:20) NewLine |\n|
+//@[000:1520) ProgramSyntax
+//@[019:0020) ├─Token(NewLine) |\n|
 param dnsPrefix string
-//@[0:22) ParameterDeclarationSyntax
-//@[0:5)  Identifier |param|
-//@[6:15)  IdentifierSyntax
-//@[6:15)   Identifier |dnsPrefix|
-//@[16:22)  SimpleTypeSyntax
-//@[16:22)   Identifier |string|
-//@[22:23) NewLine |\n|
+//@[000:0022) ├─ParameterDeclarationSyntax
```

Comment:
The character index seems to have too many digits; is the padding determined by the number of characters in the file (not the character count of the longest line)?

Reply:
The padding is determined by the longest 'start char' and the longest 'end char'.

The reason for this large amount of padding is - for start char, we've got a piece of syntax which starts at char 103 (3 digits), and for end char the ProgramSyntax ends at char 1520 (4 digits).

Previously we weren't including the ProgramSyntax in this file - I added it in as the root node of the tree as it felt useful for understanding of the syntax tree, but it does add this quirk as it spans the entire file.

I was also thinking about padding with spaces instead of zeros, but IMO this looks strange:
```
//@[000:0022) ├─ParameterDeclarationSyntax
```
vs
```
//@[  0:  22) ├─ParameterDeclarationSyntax
```

Reply:
I agree with your decision to go with zeros instead of spaces. I must have glossed over line 2 and instead looked at `AKS_LF/main.bicep` (the longest line in which is [104 characters](https://github.com/Azure/bicep/blob/af62b2859bc34cf0985c948f741d6aa473c94b03/src/Bicep.Core.Samples/Files/AKS_LF/main.bicep#L25)).

Is it odd that the starting char is always an index on the line to which the comment applies, but the end char may not be on the same line? I can't think of a more intuitive way to label the start and end indices given that some syntax units will span multiple lines, but it wasn't something I immediately picked up on.

Reply:
> Is it odd that the starting char is always an index on the line to which the comment applies, but the end char may not be on the same line?

Yeah that's something I thought about a lot when I first added these files - I didn't come up with any great ideas to make it intuitive. It's also a bit of a problem that the use of numbers isn't particularly review-able - it would be easy to miss the span being off by one, or covering too many/too few characters.